### PR TITLE
Staking account action page working

### PR DESCRIPTION
**Pull request by KarishmaBothara**
Used the new derived query from cennznet API to have the staking action work. Attaching the gif of the actions performed on this page.
staking page

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -2323,6 +2313,14 @@
   dependencies:
     "@types/node" ">= 8"
 
+"@plugnet/doughnut-wasm@^0.2.0":
```
Comment (KarishmaBothara) [inline yarn.lock:2316]: not sure..why @plugnet/doughnut-wasm@^0.2.0 appears in lock file.. can revert the lock file if required..

File: packages/app-staking/src/Actions/Account/index.tsx [hunk 2]
```diff
@@ -46,23 +47,25 @@ interface StakeState {
...
     : null;
 }
 
-function getStakeState (allAccounts: string[], allStashes: string[] | undefined, { controllerId: _controllerId, nextSessionIds, nominators, rewardDestination, sessionIds, stakers, stakingLedger, validatorPrefs }: DerivedStakingAccount, stashId: string, validateInfo: ValidatorInfo): StakeState {
+function getStakeState (allAccounts: string[], allStashes: string[] | undefined, { controllerId: _controllerId, nextSessionKeys, nominators, rewardDestination, sessionKeys, stakers, stakingLedger, validatorPrefs }: DerivedStakingAccount, stashId: string, validateInfo: ValidatorInfo): StakeState {
   const isStashNominating = !!(nominators?.length);
-  const isStashValidating = !validateInfo[1].isEmpty || !!allStashes?.includes(stashId);
-  const nextConcat = u8aConcat(...nextSessionIds.map((id): Uint8Array => id.toU8a()));
-  const currConcat = u8aConcat(...sessionIds.map((id): Uint8Array => id.toU8a()));
+  const isStashValidating = !validateInfo.isEmpty || !!allStashes?.includes(stashId);
```
Comment (holygits) [inline packages/app-staking/src/Actions/Account/index.tsx:66]: Why was this change required: `!validateInfo[1].isEmpty` -> `!validateInfo.isEmpty` ? 💭

File: packages/app-staking/src/Actions/Account/index.tsx [hunk 3]
```diff
@@ -94,8 +97,10 @@ function Account ({ allStashes, className, isOwnStash, next, onUpdateType, staki
   const validateInfo = useCall<ValidatorInfo>(api.query.staking.validators, [stashId]);
   const stakingAssetId = useCall<AssetId>(api.query.genericAsset.stakingAssetId, []);
   const freeBalance = useCall<Balance>(api.query.genericAsset.freeBalance, [stakingAssetId, stashId]);
-  const stakingAccount = useCall<DerivedStakingAccount>(api.derive.staking.account as any, [stashId]);
-  const [{ controllerId, destination, hexSessionIdQueue, hexSessionIdNext, isLoading, isOwnController, isStashNominating, isStashValidating, nominees, sessionIds, validatorPrefs }, setStakeState] = useState<StakeState>({ controllerId: null, destination: 0, hexSessionIdNext: null, hexSessionIdQueue: null, isLoading: true, isOwnController: false, isStashNominating: false, isStashValidating: false, sessionIds: [] });
+  const stakingAccountInfo = useCall<DerivedStakingInfo>((api.derive.staking as any).accountInfo, [stashId]);
+  const stakingSessionDetails = useCall<DerivedSessionKeyInfo>((api.derive.session as any).keyInfo, [stashId]);
```
Comment (holygits) [inline packages/app-staking/src/Actions/Account/index.tsx:101]: For consistency,
```suggestion
  const stakingSessionInfo = useCall<DerivedSessionKeyInfo>((api.derive.session as any).keyInfo, [stashId]);
```

**Review by holygits — APPROVED** [review]
couple nits to resolve but otherwise lgtm!

File: yarn.lock (hunk 1 above)
Comment (holygits) [reply yarn.lock:2316]: yeah it might be updated from older work on this branch

File: packages/app-staking/src/Actions/Account/index.tsx (hunk 2 above)
Comment (KarishmaBothara) [reply packages/app-staking/src/Actions/Account/index.tsx:66]: api.query.staking.nominators for a stashId will always return me Option of Map and not Option of array.. Map is a structure where I can check if it is empty or not..

File: packages/app-staking/src/Actions/Account/index.tsx (hunk 3 above)
Comment (KarishmaBothara) [reply packages/app-staking/src/Actions/Account/index.tsx]: done

File: yarn.lock (hunk 1 above)
Comment (KarishmaBothara) [reply yarn.lock]: I removed this part from the yarn.lock file..
